src/query/graphite/native/builtin_functions.go

```diff
@@ -1771,12 +1771,16 @@ func substr(_ *common.Context, seriesList singlePathSpec, start, stop int) (ts.S
 		numParts := len(nameParts)
 		currStart, currStop := start, stop
 		// Graphite supports negative indexing, so we need to also.
-		if currStart < 0 {
-			currStart += numParts
-		}
-		if currStop < 0 {
-			currStop += numParts
+		if numParts > 0 {
```

Comment:
Hm, Don't think it's possible to ever have `numParts` be less than or equal to zero?

Even with zero occurrences of the substring you always get 1 back since it's just the original input if the substring wasn't found:
https://play.golang.org/p/L3xEX7rIT3D